c_src/tstack.h

```diff
@@ -0,0 +1,39 @@
+#ifndef TSTACK_H
+#define TSTACK_H
+
+#include "ei.h"
+#define GROW_BY 30
+
+typedef struct {
+  ETERM* *data;
+  size_t used;
+  size_t size;
+} tstack;
+
+void tstack_init(tstack *stack, size_t initial_size) {
```

Comment:
Can you move the implementation of each function to `c_src/tstack.c` and keep the headers in here?